base/broadcast.jl

```diff
@@ -280,7 +280,10 @@ end
 eltypestuple(a) = (Base.@_pure_meta; Tuple{eltype(a)})
 eltypestuple(T::Type) = (Base.@_pure_meta; Tuple{Type{T}})
 eltypestuple(a, b...) = (Base.@_pure_meta; Tuple{eltypestuple(a).types..., eltypestuple(b...).types...})
-_broadcast_eltype(f, A, Bs...) = Base._return_type(f, eltypestuple(A, Bs...))
```

Comment:
I guess I could do that, but that would mean `_broadcast_eltype` doesn't actually return the type used by `broadcast`. This can be confusing, and possibly slightly annoying for packages which call this function for their `broadcast` implementation. What's the advantage of doing that?